base/inference.jl

```diff
@@ -399,7 +396,11 @@ const apply_type_tfunc = function (A, args...)
         largs == 2 && return args[2]
         args = args[2:end]
         if all(isType, args)
-            return Type{Union{map(t->t.parameters[1],args)...}}
+            try
+                return Type{Union{map(t->t.parameters[1],args)...}}
```

Comment:
Under what circumstances does this now throw and do we also need `_all` in analogy with `_any` above?